### PR TITLE
[CI] Support python package publish

### PR DESCRIPTION
**Pull request by brosoul**
## Pull Request Description
- Add some information needed on PyPI in `pyproject.toml`
- Add workflow to publish package
## Related Issues
Resolves: #124 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Upload Python Package
+
+on:
+  release:
+    types: [published]
```
Comment (brosoul) [inline .github/workflows/python-publish.yml:5]: Due to the inability to repeatedly upload packages with the same version name on PyPI (even if the previous packages with the same name are manually deleted), the triggering conditions for this workflow need to be cautious. Or do you have any other better suggestions? @Jeffwan

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install poetry
+        run: |
+          cd python/aibrix
+          python -m pip install --upgrade pip
+          pip install -U pip poetry
+      - name: Build and publish
+        env:
+          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYPI_TOKEN }}
```
Comment (brosoul) [inline .github/workflows/python-publish.yml:28]: We need to set up Actions secrets on GitHub here, and we can discuss it offline. @Jeffwan

File: python/aibrix/pyproject.toml [hunk 3]
```diff
@@ -1,12 +1,37 @@
 [tool.poetry]
 name = "aibrix"
-version = "0.1.0"
-description = ""
-authors = ["brosoul <brosoul@126.com>"]
+version = "0.1.0-rc.1"
+description = "AI Runtime for aibrix"
+authors = [
+    "AIBrix Authors <brosoul@126.com>"
```
Comment (brosoul) [inline python/aibrix/pyproject.toml:6]: Do we need a group email here

File: .github/workflows/python-publish.yml (hunk 2 above)
Comment (Jeffwan) [reply .github/workflows/python-publish.yml:28]: Sounds good, share the key with me offline and let me add it in the github setting

File: python/aibrix/pyproject.toml (hunk 3 above)
Comment (Jeffwan) [reply python/aibrix/pyproject.toml:6]: Let me create one later. We already have the domain and email should be easy

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (Jeffwan) [reply .github/workflows/python-publish.yml:5]: `published` covers the `pre-releases` phase and I think that's the right type to use.
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release

File: python/aibrix/pyproject.toml [hunk 4]
```diff
@@ -1,12 +1,37 @@
 [tool.poetry]
 name = "aibrix"
-version = "0.1.0"
-description = ""
-authors = ["brosoul <brosoul@126.com>"]
+version = "0.1.0-rc.1"
+description = "AI Runtime for aibrix"
```
Comment (Jeffwan) [inline python/aibrix/pyproject.toml]: python package is not just ai runtime. we may have other features. for example, cli etc. let's use project description

File: python/aibrix/pyproject.toml (hunk 3 above)
Comment (Jeffwan) [reply python/aibrix/pyproject.toml:6]: please help create an issue to track it. This is not v0.1.0 release blocker item. We can make it later

Comment (brosoul) [reply python/aibrix/pyproject.toml:6]: Issue record here #148

File: python/aibrix/pyproject.toml (hunk 4 above)
Comment (brosoul) [reply python/aibrix/pyproject.toml]: Done

**Pull request merged**